File: dry_cleaning_system_HmOS/entry/src/main/ets/models/Order.ts
/**
 * 订单数据模型
 */
export interface Order {
  id: number;
  orderNo: string;
  customerId: number;
  totalPrice: number;
  prepaid: number;
  payType: string;
  urgent: number;
  status: string;
  expectedTime: string;
  createTime: string;
}

/**
 * 创建订单请求
 */
export interface CreateOrderRequest {
  orderNo: string;
  customerId: number;
  totalPrice: number;
  prepaid?: number;
  payType: string;
  urgent?: number;
  status?: string;
  expectedTime?: string;
  createTime?: string;
}

/**
 * 衣物数据模型
 */
export interface Clothes {
  id: number;
  orderId: string;
  type: string;
  price: number;
  damageRemark: string;
  damageImage: string;
  status: string;
  createTime: string;
}

/**
 * 创建衣物请求
 */
export interface CreateClothesRequest {
  orderId: string;
  type: string;
  price: number;
  damageRemark?: string;
  damageImage?: string;
  status?: string;
}

File: dry_cleaning_system_Android/app/src/main/java/com/chaser/drycleaningsystem/data/entity/Order.kt
package com.chaser.drycleaningsystem.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * 订单实体类
 */
@Entity(tableName = "orders")
data class Order(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    @ColumnInfo(name = "order_no") val orderNo: String,
    @ColumnInfo(name = "customer_id") val customerId: Long,
    @ColumnInfo(name = "total_price") val totalPrice: Double,
    val prepaid: Double = 0.0,
    @ColumnInfo(name = "pay_type") val payType: String,
    val urgent: Int = 0,
    val status: String,
    @ColumnInfo(name = "expected_time") val expectedTime: String?,
    @ColumnInfo(name = "create_time") val createTime: String = System.currentTimeMillis().toString()
)
